include/bluetooth/bluetooth.h

```diff
@@ -443,6 +443,12 @@ struct bt_le_scan_param {
 	u16_t window;
 };
 
+/** Listener context for (LE) scanning. */
+struct bt_le_scan_listener {
+	bt_le_scan_cb_t *recv; /**< Callback to call on every scan packet. */
```

Comment:
We might have to consider how this will look with AE changes. Because AE would add more information in the advertising reports. As well as introducing a possible scan_stopped callback.

My suggestion would be to replace the rssi parameter with a pointer to struct containing additional information, which for now would only be the RSSI.
```
struct bt_le_adv_info {
	s8_t rssi;

        // possible additional ADV_EXT parameters:
	const bt_addr_le_t *direct_addr;

	u8_t primary_phy;
	u8_t secondary_phy;


	s8_t tx_power;

	u16_t per_interval;
}

/** Listener context for (LE) scanning. */
struct bt_le_scan_cb {
	void (*recv) (const bt_addr_le_t *addr, struct bt_le_adv_info *info,
		     u8_t adv_type, struct net_buf_simple *buf);

	void (*timeout)(void);

	sys_snode_t node;
};
```

Reply:
Seems reasonable, but perhaps this should be part of the AE PR instead, as this function pointer type is used for both the (old) static callback and the new scanner callbacks. This PR doesn't break the API at the moment, so if we have to break it for AE anyway, it'd be better to wait with this change IMO

Reply:
Yeah, so I meant to use two different callback types, the old ones would continue to use the old typedef. And this listener interface would use a new one.
Then this PR would not break with the API.
And the AE API would not break the API.

Reply:
Might seem a bit strange. But it is a way to keep it API compatible.

Reply:
Alright, sure. I'll stick all the meta parameters in the info struct then